src/lcm/input_processing/discrete_state_conversion.py

```diff
@@ -0,0 +1,244 @@
+from collections.abc import Callable
+from dataclasses import dataclass, field, make_dataclass
+
+import jax.numpy as jnp
+from dags.signature import with_signature
+from jax import Array
+
+from lcm.functools import all_as_kwargs
+from lcm.grids import DiscreteGrid
+from lcm.input_processing.util import (
+    get_gridspecs,
+)
+from lcm.typing import ParamsDict
+from lcm.user_model import Model
+
+
+@dataclass
+class DiscreteStateConverter:
```

Comment:
```suggestion
class DiscreteGridConverter:
```

- Adjust docstring, making clearer what kinds of stuff is affected here. 
- Add test case with simulation results, these should have the user-supplied codes also for choices.